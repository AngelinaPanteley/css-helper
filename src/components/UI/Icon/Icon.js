import React from 'react';

import styles from './Icon.scss';

const icon = (props) => {
  return (
    <i className={`${styles.icon} ${styles['icon_' + props.icon]}`} />
  )
}

export default icon;