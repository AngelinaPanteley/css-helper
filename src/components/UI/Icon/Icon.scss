@font-face {
  font-family: 'css-helper';
  src: url('../../../assets/icons/font/css-helper.eot?61525200');
  src: url('../../../assets/icons/font/css-helper.eot?61525200#iefix') format('embedded-opentype'),
       url('../../../assets/icons/font/css-helper.woff2?61525200') format('woff2'),
       url('../../../assets/icons/font/css-helper.woff?61525200') format('woff'),
       url('../../../assets/icons/font/css-helper.ttf?61525200') format('truetype'),
       url('../../../assets/icons/font/css-helper.svg?61525200#css-helper') format('svg');
  font-weight: normal;
  font-style: normal;
}
/* Chrome hack: SVG is rendered more smooth in Windozze. 100% magic, uncomment if you need it. */
/* Note, that will break hinting! In other OS-es font will be not as sharp as it could be */
/*
@media screen and (-webkit-min-device-pixel-ratio:0) {
  @font-face {
    font-family: 'css-helper';
    src: url('../font/css-helper.svg?61525200#css-helper') format('svg');
  }
}
*/
 
.icon:before {
  font-family: "css-helper";
  font-style: normal;
  font-weight: normal;
  speak: none;
 
  display: inline-block;
  text-decoration: inherit;
  width: 1em;
  margin-right: .2em;
  text-align: center;
  /* opacity: .8; */
 
  /* For safety - reset parent styles, that can break glyph codes*/
  font-variant: normal;
  text-transform: none;
 
  /* fix buttons height, for twitter bootstrap */
  line-height: 1em;
 
  /* Animation center compensation - margins should be symmetric */
  /* remove if not needed */
  margin-left: .2em;
 
  /* you can be more comfortable with increased icons size */
  /* font-size: 120%; */
 
  /* Font smoothing. That was taken from TWBS */
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
 
  /* Uncomment for 3D effect */
  /* text-shadow: 1px 1px 1px rgba(127, 127, 127, 0.3); */
}
 
.icon_down:before { content: '\e800'; } /* '' */
.icon_left:before { content: '\e801'; } /* '' */
.icon_right:before { content: '\e802'; } /* '' */
.icon_up:before { content: '\e803'; } /* '' */
.icon_edit:before { content: '\e804'; } /* '' */
.icon_delete:before { content: '\e805'; } /* '' */
.icon_plus:before { content: '\e806'; } /* '' */
.icon_minus:before { content: '\e807'; } /* '' */
.icon_download:before { content: '\e808'; } /* '' */
.icon_upload:before { content: '\e809'; } /* '' */
.icon_menu:before { content: '\f008'; } /* '' */
.icon_copy:before { content: '\f0c5'; } /* '' */
.icon_account:before { content: '\f2be'; } /* '' */