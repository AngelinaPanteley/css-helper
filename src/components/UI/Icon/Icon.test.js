"use strict";

import React from 'react';
import renderer from 'react-test-renderer';
import { expect as chaiExpect } from 'chai';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16.1';
import styles from './Icon.scss';
import Icon from './Icon.js';

configure({ adapter: new Adapter() });

describe('>>>Icon', () => {

  //SNAPSHOTS

  const props = {
    icon: 'edit',
  };

  const iconComponent = <Icon {...props} />;

  const component = renderer.create(iconComponent);

  it('renders itself correctly', () => {
    let componentTree = component.toJSON();
    expect(componentTree).toMatchSnapshot();
  });

  //SHALLOW RENDERING

  let wrapper;

  beforeEach(() => {
    wrapper = shallow(iconComponent);
  });

  it('renders correct icon', () => {
    chaiExpect(wrapper.contains(
      <i className={`${styles.icon} ${styles['icon_' + props.icon]}`} />
    )).to.equal(true);
  });
})