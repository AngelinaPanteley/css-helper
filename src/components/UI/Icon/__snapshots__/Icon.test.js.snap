// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`>>>Icon renders itself correctly 1`] = `
<i
  className="undefined undefined"
/>
`;
